version.go

```diff
@@ -291,6 +292,31 @@ func (v *Version) Compare(o *Version) int {
 	return comparePrerelease(ps, po)
 }
 
+// UnmarshalJSON implements JSON.Unmarshaler interface.
+func (v *Version) UnmarshalJSON(b []byte) error {
```

Comment:
The JSON handling should be part of a different pull request. How is this different from #45? I'd like to consider JSON separately from a CLI.